app/js/docker/dockerService.js

```diff
@@ -43,24 +71,32 @@ function dockerService($http, actions, flux, alertService, configService) {
 
     /*
      * docker()
-     * @param {string, required} method: HTTP verb
-     * @param {string, required} url: Docker API URL
-     * @param {string, required} action: application action to dispatch with response data
-     * @param {string, required} host: lighthouse alias for targeted Docker instance
-     * @param {string} cid: Docker generated id for image or container
-     * @param {object} data: if method is GET, set data as query parameters,
-     *                       if POST, set in request body under 'Payload' key
+     * Main entrypoint.
+     * Prepares a Docker request and dispatches the appropriate flux action upon success.
      *
-     * Note: The exposed interface is shortened to just (host, id, data)
-     * via a partial function application.
+     * @param {string, required} namespace: namespace representation of a Docker API call (i.e. 'containers.inspect')
+     * @param {object, required} request: object representation of a Docker API call.
+     * Allowed keys:
+     *      {string, required} host: lighthouse alias for targeted Docker instance
+     *      {string} id: image or container id (generated by Docker)
+     *      {object} query: map of query parameter keys and values
+     *      {*} data: body data for POST/PUT requests
      */
-    function docker(method, url, action, host, cid, data) {
+    function d(namespace, request) {
+        var ns = namespace.split('.');
+        var endpoint = endpoints[ns[0]][ns[1]];
+
+        if (endpoint === undefined) {
+            // TODO
+            // call to alertService with a generic 'Something broke.'?
```

Comment:
Open for suggestions here. Since the namespace parameter is in no way user provided data, this shouldn't really happen. I'm thinking we could just throw up an alert with a link to the github issues page? And say something like "A critical service has failed. Please submit a bug report here." (where "here" is a link) ?